### DISCUSSION TITLE
Site-to-Site IPsec VPN between two Linux Machine (rocky to rocky linux)

### DISCUSSION THREAD

**Discussion by mehwishIftikhar** [strongswan/strongswan#2544] [Q&A]
i am using strongswan vpn for vpn connectivity. version is strongSwan U5.9.10/K4.18.0-553.27.1.el8_10.x86_64

vpn server public ip is 198.13.54.13 vpn client public ip is 104.156.239.152 and both server ipsec.secret file have the same PSK. but tunnel is not created successfullty. strongswan status showing 0 up 1 connecting i am new in vpn please guide me what is the problem in my configuration.

my strongswan configuration files for server and client is: 

vpn server:
# cat /etc/strongswan/ipsec.conf
config setup
        charondebug="all"
        uniqueids=yes
conn ateway1-to-gateway2
        type=tunnel
        auto=start
        keyexchange=ikev2
        authby=secret
        left=198.13.54.13
        leftsubnet=10.10.1.0/24
        right=104.156.239.152
        rightsubnet=10.20.1.0/24
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1!
        aggressive=no
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart
vpn client
# cat /etc/strongswan/ipsec.conf
config setup
        charondebug="all"
        uniqueids=yes
conn 2gateway-to-gateway1
        type=tunnel
        auto=start
        keyexchange=ikev2
        authby=secret
        left=104.156.239.152
        leftsubnet=10.20.1.0/24
        right=198.13.54.13
        rightsubnet=10.10.1.0/24
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1!
        aggressive=no
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart

I am trying to connect my vpn client to vpn server but it is not working. 

[root@VPN-Server ~]# strongswan statusall
Status of IKE charon daemon (strongSwan 5.9.10, Linux 4.18.0-553.27.1.el8_10.x86_64, x86_64):
  uptime: 75 minutes, since Nov 21 09:19:15 2024
  malloc: sbrk 1871872, mmap 0, used 962576, free 909296
  worker threads: 12 of 16 idle, 4/0/0/0 working, job queue: 0/0/0/0, scheduled: 1
  loaded plugins: charon pkcs11 aesni aes des rc2 sha2 sha1 md4 md5 mgf1 random nonce x509 revocation constraints acert pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl gcrypt pkcs8 fips-prf gmp curve25519 chapoly xcbc cmac hmac kdf ctr ccm gcm drbg newhope curl attr kernel-netlink resolve socket-default farp stroke vici updown eap-identity eap-sim eap-aka eap-aka-3gpp eap-aka-3gpp2 eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap xauth-generic xauth-eap xauth-pam xauth-noauth dhcp led duplicheck unity counters
Listening IP addresses:
  198.13.54.13
  10.10.1.1
  2001:19f0:7001:2b0d:5400:5ff:fe29:a9a6
Connections:
ateway1-to-gateway2:  198.13.54.13...104.156.239.152  IKEv2, dpddelay=30s
ateway1-to-gateway2:   local:  [198.13.54.13] uses pre-shared key authentication
ateway1-to-gateway2:   remote: [104.156.239.152] uses pre-shared key authentication
ateway1-to-gateway2:   child:  10.10.1.0/24 === 10.20.1.0/24 TUNNEL, dpdaction=start
Security Associations (0 up, 1 connecting):
ateway1-to-gateway2[1]: CONNECTING, 198.13.54.13[%any]...104.156.239.152[%any]
ateway1-to-gateway2[1]: IKEv2 SPIs: 71f12145e1d7a813_i* 0000000000000000_r
ateway1-to-gateway2[1]: Tasks active: IKE_VENDOR IKE_INIT IKE_NATD IKE_CERT_PRE IKE_AUTH IKE_CERT_POST IKE_CONFIG IKE_AUTH_LIFETIME IKE_MOBIKE IKE_ESTABLISH CHILD_CREATE 

[root@VPN-Server ~]# iptables -nvL
Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       104.156.239.152      198.13.54.13        

Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       10.10.1.2            10.20.1.10          
    0     0 ACCEPT     all  --  *      *       10.10.1.1            10.20.1.1

**Comment by tobiasbrunner** [comment]
Check the log of the responder. If it does not receive the IKE_SA_INIT packet, there is a firewall/network issue somewhere.

**Reply by mehwishIftikhar** [reply]
Nov 21 10:49:42 vpn-client systemd[1]: Starting strongSwan IPsec IKEv1/IKEv2 daemon using swanctl...
Nov 21 10:49:42 vpn-client charon-systemd[6713]: PKCS11 module '<name>' lacks library path
Nov 21 10:49:42 vpn-client charon-systemd[6713]: TPM 2.0 - could not load "libtss2-tcti-tabrmd.so.0"
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tpm': failed to load - tpm_plugin_create returned NULL
Nov 21 10:49:42 vpn-client charon-systemd[6713]: OpenSSL FIPS mode(0) - disabled
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'kernel-libipsec': failed to load - kernel_libipsec_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: using '/sbin/resolvconf' to install DNS servers
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'eap-tnc': failed to load - eap_tnc_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnc-ifmap': failed to load - tnc_ifmap_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnc-pdp': failed to load - tnc_pdp_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnc-imc': failed to load - tnc_imc_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnc-imv': failed to load - tnc_imv_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnc-tnccs': failed to load - tnc_tnccs_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnccs-20': failed to load - tnccs_20_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnccs-11': failed to load - tnccs_11_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: plugin 'tnccs-dynamic': failed to load - tnccs_dynamic_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client charon-systemd[6713]: unable to bind socket: Address already in use
Nov 21 10:49:42 vpn-client charon-systemd[6713]: could not open IPv6 socket, IPv6 disabled
Nov 21 10:49:42 vpn-client charon-systemd[6713]: unable to bind socket: Address already in use
Nov 21 10:49:42 vpn-client charon-systemd[6713]: could not open IPv4 socket, IPv4 disabled
Nov 21 10:49:42 vpn-client charon-systemd[6713]: could not create any sockets
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loading ca certificates from '/etc/strongswan/ipsec.d/cacerts'
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loading aa certificates from '/etc/strongswan/ipsec.d/aacerts'
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loading ocsp signer certificates from '/etc/strongswan/ipsec.d/ocspcerts'
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loading attribute certificates from '/etc/strongswan/ipsec.d/acerts'
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loading crls from '/etc/strongswan/ipsec.d/crls'
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loading secrets from '/etc/strongswan/ipsec.secrets'
Nov 21 10:49:42 vpn-client charon-systemd[6713]:   loaded IKE secret for 104.156.239.152  198.13.54.13
Nov 21 10:49:42 vpn-client charon-systemd[6713]: opening triplet file /etc/strongswan/ipsec.d/triplets.dat failed: No such file or directory
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loaded 0 RADIUS server configurations
Nov 21 10:49:42 vpn-client charon-systemd[6713]: HA config misses local/remote address
Nov 21 10:49:42 vpn-client charon-systemd[6713]: no script for ext-auth script defined, disabled
Nov 21 10:49:42 vpn-client charon-systemd[6713]: loaded plugins: charon-systemd pkcs11 aesni aes des rc2 sha2 sha1 md4 md5 mgf1 random nonce x509 revocation constraints acert pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl gcrypt>
Nov 21 10:49:42 vpn-client charon-systemd[6713]: dropped capabilities, running as uid 0, gid 0
Nov 21 10:49:42 vpn-client charon-systemd[6713]: spawning 16 worker threads
Nov 21 10:49:42 vpn-client charon-systemd[6713]: no socket implementation registered, receiving failed
Nov 21 10:49:42 vpn-client swanctl[6732]: plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
Nov 21 10:49:42 vpn-client swanctl[6732]: no files found matching '/etc/strongswan/swanctl/conf.d/*.conf'
Nov 21 10:49:42 vpn-client swanctl[6732]: no authorities found, 0 unloaded
Nov 21 10:49:42 vpn-client swanctl[6732]: no pools found, 0 unloaded
Nov 21 10:49:42 vpn-client swanctl[6732]: no connections found, 0 unloaded
Nov 21 10:49:42 vpn-client systemd[1]: Started strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.

**Reply by mehwishIftikhar** [reply]
firewall is stopped on both servers

**Reply by tobiasbrunner** [reply]
> Nov 21 10:49:42 vpn-client charon-systemd[6713]: unable to bind socket: Address already in use
> Nov 21 10:49:42 vpn-client charon-systemd[6713]: could not open IPv6 socket, IPv6 disabled
> Nov 21 10:49:42 vpn-client charon-systemd[6713]: unable to bind socket: Address already in use
> Nov 21 10:49:42 vpn-client charon-systemd[6713]: could not open IPv4 socket, IPv4 disabled
> Nov 21 10:49:42 vpn-client charon-systemd[6713]: could not create any sockets

This is your problem. Looks like some other service is running on the same ports (since you use charon-systemd, make sure you don't have the charon daemon running via ipsec/starter).

**Reply by mehwishIftikhar** [reply]
[root@VPN-Server ~]# ps aux | grep charon
root       11870  0.6  1.8 945104 14340 ?        Ssl  04:52   0:00 /usr/sbin/charon-systemd
root       11907  0.0  0.1  12216  1088 pts/2    R+   04:52   0:00 grep --color=auto charon
[root@VPN-Server ~]# sudo ss -tuln | grep -E '500|4500'
udp   UNCONN 0      0            0.0.0.0:4500      0.0.0.0:*          
udp   UNCONN 0      0            0.0.0.0:500       0.0.0.0:*          
udp   UNCONN 0      0               [::]:4500         [::]:*          
udp   UNCONN 0      0               [::]:500          [::]:*

[root@vpn-client ~]# ps aux | grep charon
root       16315  0.1  1.7 945104 13828 ?        Ssl  05:08   0:00 /usr/sbin/charon-systemd
root       16352  0.0  0.1  12216  1112 pts/2    S+   05:08   0:00 grep --color=auto charon
[root@vpn-client ~]# ss -tuln | grep -E '500|4500'
udp   UNCONN 0      0            0.0.0.0:4500      0.0.0.0:*          
udp   UNCONN 0      0            0.0.0.0:500       0.0.0.0:*          
udp   UNCONN 0      0               [::]:4500         [::]:*          
udp   UNCONN 0      0               [::]:500          [::]:* 

only charon-systemd is running 
but when we check strongswan status then its empty nothing connection and up?

# journalctl -u strongswan
Nov 22 05:17:33 VPN-Server systemd[1]: Stopping strongSwan IPsec IKEv1/IKEv2 daemon using swanctl...
Nov 22 05:17:33 VPN-Server charon-systemd[11870]: SIGTERM received, shutting down
Nov 22 05:17:33 VPN-Server systemd[1]: strongswan.service: Succeeded.
Nov 22 05:17:33 VPN-Server systemd[1]: Stopped strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.
Nov 22 05:17:33 VPN-Server systemd[1]: Starting strongSwan IPsec IKEv1/IKEv2 daemon using swanctl...
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: PKCS11 module '<name>' lacks library path
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: TPM 2.0 - could not load "libtss2-tcti-tabrmd.so.0"
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tpm': failed to load - tpm_plugin_create returned NULL
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: OpenSSL FIPS mode(0) - disabled
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'kernel-libipsec': failed to load - kernel_libipsec_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: using '/sbin/resolvconf' to install DNS servers
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'eap-tnc': failed to load - eap_tnc_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnc-ifmap': failed to load - tnc_ifmap_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnc-pdp': failed to load - tnc_pdp_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnc-imc': failed to load - tnc_imc_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnc-imv': failed to load - tnc_imv_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnc-tnccs': failed to load - tnc_tnccs_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnccs-20': failed to load - tnccs_20_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnccs-11': failed to load - tnccs_11_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: plugin 'tnccs-dynamic': failed to load - tnccs_dynamic_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loading ca certificates from '/etc/strongswan/ipsec.d/cacerts'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loading aa certificates from '/etc/strongswan/ipsec.d/aacerts'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loading ocsp signer certificates from '/etc/strongswan/ipsec.d/ocspcerts'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loading attribute certificates from '/etc/strongswan/ipsec.d/acerts'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loading crls from '/etc/strongswan/ipsec.d/crls'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loading secrets from '/etc/strongswan/ipsec.secrets'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]:   loaded IKE secret for 198.13.54.13  104.156.239.152
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: opening triplet file /etc/strongswan/ipsec.d/triplets.dat failed: No such file or directory
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loaded 0 RADIUS server configurations
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: HA config misses local/remote address
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: no script for ext-auth script defined, disabled
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loaded plugins: charon-systemd pkcs11 aesni aes des rc2 sha2 sha1 md4 md5 mgf1 random nonce x509 revocation constraints acert pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl gcryp>
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: dropped capabilities, running as uid 0, gid 0
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: spawning 16 worker threads
Nov 22 05:17:33 VPN-Server swanctl[12627]: plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
Nov 22 05:17:33 VPN-Server swanctl[12627]: no files found matching '/etc/strongswan/swanctl/conf.d/*.conf'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: loaded IKE shared key with id 'ike-1' for: '%any'
Nov 22 05:17:33 VPN-Server charon-systemd[12607]: added vici connection: myvpn
Nov 22 05:17:33 VPN-Server swanctl[12627]: loaded ike secret 'ike-1'
Nov 22 05:17:33 VPN-Server swanctl[12627]: no authorities found, 0 unloaded
Nov 22 05:17:33 VPN-Server swanctl[12627]: no pools found, 0 unloaded
Nov 22 05:17:33 VPN-Server swanctl[12627]: loaded connection 'myvpn'
Nov 22 05:17:33 VPN-Server swanctl[12627]: successfully loaded 1 connections, 0 unloaded
Nov 22 05:17:33 VPN-Server systemd[1]: Started strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.


please guide me...

**Reply by mehwishIftikhar** [reply]
my configuration is 
vpn server 
[root@VPN-Server strongswan]# cat /etc/strongswan/ipsec.conf
config setup
        charondebug="all"
        uniqueids=yes
conn myvpn 
        type=tunnel
        auto=start
        keyexchange=ikev2
        authby=secret
        left=198.13.54.13
        leftsubnet=10.10.1.0/24
        right=104.156.239.152
        rightsubnet=10.20.1.0/24
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1!
        aggressive=no
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart
vpn client 
[root@vpn-client ~]# cat /etc/strongswan/ipsec.conf
config setup
        charondebug="all"
        uniqueids=yes
conn myvpn
        type=tunnel
        auto=start
        keyexchange=ikev2
        authby=secret
        left=104.156.239.152
        leftsubnet=10.20.1.0/24
        right=198.13.54.13
        rightsubnet=10.10.1.0/24
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1!
        aggressive=no
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart
after starting the service i checked the strongswan status 

# strongswan statusall
Status of IKE charon daemon (strongSwan 5.9.10, Linux 4.18.0-553.27.1.el8_10.x86_64, x86_64):
  uptime: 2 minutes, since Nov 22 07:08:18 2024
  malloc: sbrk 1871872, mmap 0, used 832672, free 1039200
  worker threads: 12 of 16 idle, 4/0/0/0 working, job queue: 0/0/0/0, scheduled: 1
  loaded plugins: charon pkcs11 aesni aes des rc2 sha2 sha1 md4 md5 mgf1 random nonce x509 revocation constraints acert pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl gcrypt pkcs8 fips-prf gmp curve25519 chapoly xcbc cmac hmac kdf ctr ccm gcm drbg newhope curl attr kernel-netlink resolve socket-default farp stroke vici updown eap-identity eap-sim eap-aka eap-aka-3gpp eap-aka-3gpp2 eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap xauth-generic xauth-eap xauth-pam xauth-noauth dhcp led duplicheck unity counters
Listening IP addresses:
  104.156.239.152
  10.20.1.1
  2001:19f0:7001:1563:5400:5ff:fe2a:7648
Connections:
       myvpn:  104.156.239.152...198.13.54.13  IKEv1/2, dpddelay=30s
       myvpn:   local:  [104.156.239.152] uses pre-shared key authentication
       myvpn:   remote: [198.13.54.13] uses pre-shared key authentication
       myvpn:   child:  10.20.1.0/24 === 10.10.1.0/24 TUNNEL, dpdaction=start
Security Associations (0 up, 1 connecting):
       myvpn[1]: CONNECTING, 104.156.239.152[%any]...198.13.54.13[%any]
       myvpn[1]: IKEv2 SPIs: 0777238121ffa5fe_i* 0000000000000000_r
       myvpn[1]: Tasks active: IKE_VENDOR IKE_INIT IKE_NATD IKE_CERT_PRE IKE_AUTH IKE_CERT_POST IKE_CONFIG IKE_AUTH_LIFETIME IKE_MOBIKE IKE_ESTABLISH CHILD_CREATE

what is the issue in my configuration and networks?
